pr-type/refactor: #8514

netvetra opened the pull request:
…ube projects list for user rights issue #8514

[Refactor][sonarqube] Can you please change API used to obtain SonarQube projects list for user rights issue #8514

Line changed: 134 only.

<!--
Licensed to the Apache Software Foundation (ASF) under one or more
contributor license agreements.  See the NOTICE file distributed with
this work for additional information regarding copyright ownership.
The ASF licenses this file to You under the Apache License, Version 2.0
(the "License"); you may not use this file except in compliance with
the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

<!--
Thanks for submitting a pull request!

We appreciate you spending the time to work on these changes.
Please fill out as many sections below as possible.
-->

### Summary
Replace /api/projects/search with /api/components/search_projects endpoint works without admin rights.

### Does this close any open issues?
Closes #8514

### Screenshots
N/A

### Other Information
N/A

netvetra committed:
[Refactor][sonarqube] Can you please change API used to obtain SonarQube projects list for user rights issue #8514

[Refactor][sonarqube] Can you please change API used to obtain SonarQube projects list for user rights issue #8514

Line changed: 134 only.

klesh reviewed (commented):
Thanks for your contribution.
I think we need to add the `qualifiers=TRK ` parameter in order for the API to return projects only according to https://next.sonarqube.com/sonarqube/web_api/api/components

netvetra committed:
Merge branch 'main' into main

netvetra committed:
Update blueprint_v200.go

append query.Set("qualifiers", "TRK")

netvetra committed:
Update remote_api.go

netvetra committed:
Update remote_api.go

netvetra committed:
Update remote_api.go

netvetra committed:
Update remote_api.go

netvetra committed:
Merge branch 'main' into main

netvetra committed:
Update blueprint_v200.go

Remove a space for:
  backend/plugins/sonarqube/api/blueprint_v200.go:134: File is not `gofmt`-ed with `-s` (gofmt)
  	query.Set("qualifiers", "TRK")

netvetra committed:
Update remote_api.go

Simply format

netvetra committed:
Merge branch 'main' into main
